application-analytics-default/src/main/java/com/xwiki/analytics/script/AnalyticsScriptService.java

```diff
@@ -75,6 +75,7 @@ public JsonNode getMatomoRequestResult(Map<String, String> parameters, Map<Strin
     /**
      *
      * @return the analytics configuration.
+     * @since 1.0 I must add the @since to be able to build the application
```

Comment:
``@Unstable`` works only if we know when the annotation target (method in this case) was introduced. A method cannot stay unstable forever. For XWiki Standard the policy is https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#H40UnstableAnnotation . For Pro Apps we could keep it until we increase the major version (i.e. go from 1.x to 2.x). But in any case we need the ``@since`` version specified when ``@Unstable`` is used.

Now, there's something I don't understand. This class is marked with ``@since 1.1`` and ``@Unstable``. So I take it that it was introduced in version 1.1 of the application, and it's unstable since then. How can this method be since version 1.0 if this class wasn't available in that version?

You should mark with ``@Unstable`` (and ``@since``) only the methods that were introduced in a **different** (future) version of the class. If a method doesn't specify the ``@since`` version and is not marked with ``@Unstable`` it means it inherits these from the class. It means that the method was introduced when the class was introduced and is unstable if the class is unstable.

I see that there's no version released for the Analytics Application, which means:
* ``@since 1.1`` on the class is wrong, it should be ``@since 1.0``
* you don't need ``@since`` and ``@Unstable`` on the method because it is introduced in the same version as the class (1.0)

Reply:
The presence of version 1.1 in the class was an oversight on my part.